src/OpenApi/test/Services/OpenApiComponentService/OpenApiComponentService.ParameterSchemas.cs

```diff
@@ -0,0 +1,389 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.OpenApi.Any;
+using Microsoft.OpenApi.Models;
+
+public partial class OpenApiComponentServiceTests : OpenApiDocumentServiceTestBase
+{
+#nullable enable
+    public static object?[][] RouteParametersWithPrimitiveTypes =>
+    [
+        [(int id) => {}, "integer", "int32", false],
+        [(long id) => {}, "integer", "int64", false],
+        [(float id) => {}, "number", "float", false],
+        [(double id) => {}, "number", "double", false],
+        [(decimal id) => {}, "number", "double", false],
+        [(bool id) => {}, "boolean", null, false],
+        [(string id) => {}, "string", null, false],
+        [(char id) => {}, "string", null, false],
+        [(byte id) => {}, "string", "byte", false],
+        [(short id) => {}, "integer", null, false],
+        [(ushort id) => {}, "integer", null, false],
+        [(uint id) => {}, "integer", "int32", false],
+        [(ulong id) => {}, "integer", "int64", false],
+        [(Uri id) => {}, "string", "uri", false],
+        [(int? id) => {}, "integer", "int32", true],
+        [(long? id) => {}, "integer", "int64", true],
+        [(float? id) => {}, "number", "float", true],
+        [(double? id) => {}, "number", "double", true],
+        [(decimal? id) => {}, "number", "double", true],
+        [(bool? id) => {}, "boolean", null, true],
+        [(string? id) => {}, "string", null, true],
+        [(char? id) => {}, "string", null, true],
+        [(byte? id) => {}, "string", "byte", true],
+        [(short? id) => {}, "integer", null, true],
+        [(ushort? id) => {}, "integer", null, true],
+        [(uint? id) => {}, "integer", "int32", true],
+        [(ulong? id) => {}, "integer", "int64", true],
+        [(Uri? id) => {}, "string", "uri", true],
+        [(TimeOnly id) => {}, "string", "time", false],
+        [(DateOnly id) => {}, "string", "date", false],
+        [(TimeOnly? id) => {}, "string", "time", true],
+        [(DateOnly? id) => {}, "string", "date", true]
```

Comment:
Isn't it usually `"date-time"`?

Reply:
The specs supports `date` as a format for date values without a time component.